src/crypto/slow-hash.c

```diff
@@ -214,6 +215,33 @@ extern void aesb_pseudo_round(const uint8_t *in, uint8_t *out, const uint8_t *ex
     lo ^= SWAP64LE(*(U64(hp_state + (j ^ 0x20)) + 1)); \
   } while (0)
 
+#define VARIANT4_RANDOM_MATH_INIT() \
+  v4_reg r[8]; \
+  struct V4_Instruction code[TOTAL_LATENCY * ALU_COUNT + 1]; \
```

Comment:
No, it never produces more than `ALU_COUNT` instructions for each cycle because it has to find a free ALU for current cycle before adding each new instruction.

Reply:
It's also never more than 69 instructions for block heights < 10,000,000 while this array has space for 135 instructions.

Reply:
Again, I guess this is acceptable because this won't use that many block heights? A check for height after adding an instruction would have minimal impact and could be used to abort the entire thing anyway.

Reply:
I don't quite get it, what additional check do you want here?

Reply:
To check for 69/70 instructions and then to immediately return.

Reply:
That is a good idea. I've added a check that generated code never has more than 70 instructions (without final RET).